### PR TITLE
VP-6942  fix for Member ModifiedDate not updating

### PR DESCRIPTION
**Pull request by ksavosteev**
Modified date (and other auditable entity properties) were not updated when saving member with only changed navigation properties. This forces change tracker to treat an entity as updated.

https://virtocommerce.atlassian.net/browse/VP-6942

### CODE REVIEW

**Comment by mvktsk** [comment]
Review task https://virtocommerce.atlassian.net/browse/VP-7134 has been created

File: src/VirtoCommerce.CustomerModule.Data/Repositories/IMemberRepository.cs [hunk 1]
```diff
@@ -13,8 +13,10 @@ public interface IMemberRepository : IRepository
         IQueryable<NoteEntity> Notes { get; }
         IQueryable<PhoneEntity> Phones { get; }
         IQueryable<MemberRelationEntity> MemberRelations { get; }
-        
+
         Task<MemberEntity[]> GetMembersByIdsAsync(string[] ids, string responseGroup = null, string[] memberTypes = null);
         Task RemoveMembersByIdsAsync(string[] ids, string[] memberTypes = null);
+
+        void SetChangedState(MemberEntity entity);
```
Comment (Vectorfield4) [inline src/VirtoCommerce.CustomerModule.Data/Repositories/IMemberRepository.cs:20]: How about `SetModifiedState`?

**Review by Vectorfield4 — APPROVED** [review]

File: src/VirtoCommerce.CustomerModule.Data/Repositories/IMemberRepository.cs (hunk 1 above)
Comment (ksavosteev) [reply src/VirtoCommerce.CustomerModule.Data/Repositories/IMemberRepository.cs:20]: yes

File: src/VirtoCommerce.CustomerModule.Data/Services/MemberService.cs [hunk 2]
```diff
@@ -142,11 +142,15 @@ public virtual async Task SaveChangesAsync(Member[] members)
...
                         {
-                           dataSourceMember.FromModel(member, pkMap);
+                            dataSourceMember.FromModel(member, pkMap);
 
                             var dataTargetMember = existingMemberEntities.FirstOrDefault(m => m.Id == member.Id);
                             if (dataTargetMember != null)
                             {
+                                /// Workaround to trigger update of audtiable fields when only updating navigation properties.
+                                /// Otherwise on update trigger is fired only when non navigation properties are updated.
+                                repository.SetChangedState(dataTargetMember);
```
Comment (Vectorfield4) [inline src/VirtoCommerce.CustomerModule.Data/Services/MemberService.cs:152]: Add unit-test to this call.

Comment (tatarincev) [reply src/VirtoCommerce.CustomerModule.Data/Services/MemberService.cs:152]: How about doing it this way?
https://github.com/VirtoCommerce/vc-module-catalog/blob/support/2.x/VirtoCommerce.CatalogModule.Data/Services/ItemServiceImpl.cs#L180

Comment (ksavosteev) [reply src/VirtoCommerce.CustomerModule.Data/Services/MemberService.cs]: fixed, unit test added

**Review by Vectorfield4 — APPROVED** [review]
100% coverage 🤗

**Pull request merged**